File: more-examples-pageobjects/java-testng/src/main/java/pageObjects/UserAccountPage.java
package pageObjects;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class UserAccountPage {

    WebDriver webDriver;

    @FindBy(css = "h3.RTL")
    WebElement welcomeMessage;

    public UserAccountPage(WebDriver webDriver) {
        this.webDriver = webDriver;
        PageFactory.initElements(this.webDriver, this);
    }

    public String getWelcomeMessage() {
        return welcomeMessage.getText();
    }
}

File: more-examples-pageobjects/js-mocha-chai/pageObjects/UserAccountPage.js
var webDriver = require('selenium-webdriver');
var By = webDriver.By;

UserAccountPage = function UserAccountPage(driver) {
    this.driver = driver;
    this.welcomeMessage = By.css("h3.RTL");
};

UserAccountPage.prototype.getWelcomeMessage = function() {
    var d = webDriver.promise.defer();
    this.driver.findElement(this.welcomeMessage).getText().then(function (text) {
        d.fulfill(text);
    });
    return d.promise;
};

module.exports = UserAccountPage;